src/commands/apphosting-config-export.ts

```diff
@@ -0,0 +1,64 @@
+import { Command } from "../command";
+import { logger } from "../logger";
+import { Options } from "../options";
+import { needProjectId } from "../projectUtils";
+import { requireAuth } from "../requireAuth";
+import * as secretManager from "../gcp/secretManager";
+import { requirePermissions } from "../requirePermissions";
+import { APPHOSTING_LOCAL_YAML, allYamlPaths, yamlPath } from "../apphosting/config";
+import { fetchSecrets, getConfigToExport } from "../apphosting/secrets";
+import { join } from "path";
+import { loadAppHostingYaml } from "../apphosting/yaml";
+
+export const command = new Command("apphosting:config:export")
+  .description(
+    "Export apphosting configurations such as secrets into an apphosting.local.yaml file",
+  )
+  .option(
+    "-s, --secrets <apphosting.yaml or apphosting.<environment>.yaml file to export secrets from>",
+    "This command combines the base apphosting.yaml with the specified environment-specific file (e.g., apphosting.staging.yaml). If keys conflict, the environment-specific file takes precedence.",
+  )
+  .before(requireAuth)
+  .before(secretManager.ensureApi)
+  .before(requirePermissions, ["secretmanager.versions.access"])
+  .action(async (options: Options) => {
+    const projectId = needProjectId(options);
+    const currentDir = process.cwd();
+
+    // Get all apphosting yaml files ignoring the apphosting.local.yaml file
+    const yamlFilePaths = allYamlPaths(currentDir)?.filter(
```

Comment:
where are we using the value of the --secrets flag?

Reply:
missed this, thanks for catching it!